src/settings.cpp

```diff
@@ -1124,6 +1137,11 @@ extern "C" int openmc_set_n_batches(
       !(contains(settings::statepoint_batch, n_batches)))
     settings::statepoint_batch.insert(n_batches);
 
+  // Add value of n_batches to surface_source_batch
+  if (add_statepoint_batch &&
```

Comment:
Let's extend the comment a bit to explain the rationale for why we're modifying the surface source batches as well here.

Reply:
I clarified that this is necessary when you don't indicate the batches where you want to save the source. However, I realized that I was using a variable that comes from the statepoint batches, so I added a particular one for surface source.